### PR TITLE
qemu: Add configuration options to specify cpu count and memory size

### LINKED ISSUE #7149: Add configuration options to qemu builder to specify cpu count and memory size
@arizvisa I noticed your pull requests which normalize how cpu/memory values are configured when using the vmware/parallels/virtualbox builders, so I'm including your name on this issue. That's because I noticed while updating my templates to the new and improved standard, that the `qemu` builder still lacks this functionality... so I thought I'd open an issue. Feel free to close this ticket if  the `qemu` builder ostracization was intentional.

### PR DESCRIPTION
**Pull request by vtolstov**
provide memory and cpus like other builders does

Signed-off-by: Vasiliy Tolstov <v.tolstov@unistack.org>

### CODE REVIEW

**Comment by ladar** [comment]
Code changes look fine, but I'm won't be able to personally test the change for a couple of weeks. I'm currently working on non-`packer` projects.

**Comment by azr** [comment]
Hello @vtolstov, thanks. Behaviour LGTM I just added some tweaks in the messages so it's clearer.
After fixing this, if someone could validate that this works by testing it we can merge I think 🙂

**Comment by ladar** [comment]
I can test the next time I kick off a roboxes build, but that probably won't be for a week or two...

**Comment by SwampDragons** [comment]
I think we'd be willing to wait for validation that this works as expected; just let us know when you get around to it.

**Comment by ladar** [comment]
Just following up to say I'm working on a robox build, and will test this PR with the qemu/libvirt boxes.

**Comment by vtolstov** [comment]
i'm add docs about new params to qemu builder and fix code suggestions and issues.

**Comment by rickard-von-essen** [comment]
(_close #7149_ is not a very useful subject.)

**Comment by vtolstov** [comment]
squash, and fix commit message

**Comment by ladar** [comment]
The min megabytes check is still using bytes:

```
if b.config.MemorySize < 10*1024*1024 {
	log.Printf("MemorySize %d is too small, using default: 512M", b.config.MemorySize)
	b.config.MemorySize = 512 * 1024 * 1024
}
```

Should be:

```
if b.config.MemorySize < 10 {
	log.Printf("MemorySize %d is too small, using default value: 512", b.config.MemorySize)
	b.config.MemorySize = 512
}
```

I removed the "M" suffix from the message, since the `packer` convention/docs is to use a numeric/integer value in the template.

**Comment by ladar** [comment]
@vtolstov looks like you also need to sign the recently added Contributor Agreement...

**Comment by vtolstov** [comment]
hm, sometimes ago i'm create prs and it merged without signing , where i can find info what i need to sign?

File: website/source/docs/builders/qemu.html.md.erb [hunk 1]
```diff
@@ -135,6 +135,8 @@ Linux server and have not enabled X11 forwarding (`ssh -X`).
     five seconds and one minute 30 seconds, respectively. If this isn't
     specified, the default is `10s` or 10 seconds.
 
+-   `cpus` (number) - The number of cpus to use when building the VM.
```
Comment (azr) [inline website/source/docs/builders/qemu.html.md.erb]:
```suggestion
-   `cpus` (number) - The number of cpus to use when building the VM.
     The default is `1` CPU.
```

File: website/source/docs/builders/qemu.html.md.erb [hunk 2]
```diff
@@ -237,6 +239,9 @@ Linux server and have not enabled X11 forwarding (`ssh -X`).
     qemu binary with the flags `-machine help` to list available types for
     your system. This defaults to `pc`.
 
+-   `memory` (number) - The amount of memory to use when building the VM
+    in megabytes.
```
Comment (azr) [inline website/source/docs/builders/qemu.html.md.erb]:
```suggestion
    in megabytes. This defaults to `512` megabytes.
```

**Pull request merged**